protostar/commands/test/environments/execution_environment.py

```diff
@@ -0,0 +1,58 @@
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from starkware.starknet.testing.objects import StarknetTransactionExecutionInfo
+from starkware.starkware_utils.error_handling import StarkException
+
+from protostar.commands.test.starkware.execution_resources_summary import (
+    ExecutionResourcesSummary,
+)
+from protostar.commands.test.starkware.test_execution_state import TestExecutionState
+from protostar.commands.test.test_context import TestContextHintLocal
+from protostar.commands.test.test_environment_exceptions import (
+    StarknetRevertableException,
+)
+from protostar.starknet.cheatable_execute_entry_point import (
+    CheatableExecuteEntryPoint,
+)
+from protostar.starknet.cheatcode_factory import CheatcodeFactory
+
+
+class ExecutionEnvironment(ABC):
+    def __init__(self, state: TestExecutionState):
+        self.state: TestExecutionState = state
+
+    @abstractmethod
+    def _cheatcode_factory(self) -> CheatcodeFactory:
+        ...
+
+    async def _invoke(self, function_name: str) -> Optional[ExecutionResourcesSummary]:
+        return await self._call(function_name)
+
+    async def invoke(self, function_name: str) -> Optional[ExecutionResourcesSummary]:
+        """
+        Final. Override ``_invoke`` if needed.
+        """
+        CheatableExecuteEntryPoint.cheatcode_factory = self._cheatcode_factory()
+        CheatableExecuteEntryPoint.custom_hint_locals = [
+            TestContextHintLocal(self.state.context)
+        ]
+
+        return await self._invoke(function_name)
+
+    async def _call(self, function_name: str) -> ExecutionResourcesSummary:
```

Comment:
I know it's not related to your code (it was like that before) but `call` means something different than `invoke` in starknet. We should't use this names interchangebly. Can we change this name to suggests it invokes test function instead of callin it?

Reply:
renamed